core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/RootVariableSource.java

```diff
@@ -388,18 +387,6 @@ private static ParentVariableType determineParentVariableType(Class<?> rootClass
         if (getAnnotation(declaringClass, memberName, PlanningVariable.class) != null) {
             return ParentVariableType.VARIABLE;
         }
-        if (getAnnotation(declaringClass, memberName, ShadowVariableLooped.class) != null) {
-            throw new IllegalArgumentException("""
-                    The source path (%s) starting from root class (%s) accesses a @%s property (%s).
-                    @%s properties cannot be used as a source, since they are not guaranteed to
-                    be updated when the supplier is called. Supplier methods are only called when
-                    none of their dependencies are looped, so reading @%s properties are not needed.
-                    Maybe remove the source path (%s) from the @%s?
-                    """.formatted(
-                    variablePath, rootClass.getCanonicalName(), ShadowVariableLooped.class.getSimpleName(),
-                    memberName, ShadowVariableLooped.class.getSimpleName(), ShadowVariableLooped.class.getSimpleName(),
-                    variablePath, ShadowSources.class.getSimpleName()));
-        }
```

Comment:
In the other PR, you say you think nobody should depend on `looped`, because it will be `false` when the supplier is called. This makes sense - that tells me we should actually _keep_ this fail-fast.